src/app/work-item/work-item.service.ts

```diff
@@ -435,16 +436,75 @@ export class WorkItemService {
    * place should be implemented
    */
   moveItem(wi: WorkItem, dir: String) {
-    //We need to call an ordering API which will store
-    //the new location for the selected work item
     let index = this.workItems.findIndex(x => x.id == wi.id);
-    let wiSplice = this.workItems.splice(index, 1);
-    if (dir == 'top') {
-      this.workItems.splice(0, 0, wi);
-    } else {
-      this.workItems.splice(this.workItems.length, 0, wi);
+    wi.attributes.nextitem = '';
+    wi.attributes.previousitem = '';
+    switch (dir){
+      case 'top':
+        //this.workItems.splice(0, 0, wi);
+        break;
+      case 'bottom':
+        //this.workItems.splice(this.workItems.length, 0, wi);
+        break;
+      case 'up':
+        if (index > 0) { //no moving of element if it is the first element
+          //move the work item up by 1. Below statement will create two elements
+          this.workItems.splice( index - 1 , 0, wi);
+          //remove the duplicate element
+          this.workItems.splice( index + 1, 1);
+          //Set the previous and next WI ids
+          if (index === 1) {
+            wi.attributes.nextitem = parseInt(this.workItems[index].id);
+          } else {
+            wi.attributes.previousitem = parseInt(this.workItems[index - 2].id);
+            wi.attributes.nextitem = parseInt(this.workItems[index].id);
+          }
+        }
+        break;
+      case 'down':
+        if ( index < (this.workItems.length - 1) ) { //no moving of elements if it is the last element
+          //move the work item down by 1. Below statement will create two elements
+          this.workItems.splice( index + 2 , 0, wi);
+          //remove the duplicate element
+          this.workItems.splice( index, 1);
+          //Set the previous and next WI ids
+          //Check if the element to move is the second last element or not
+          //and set the previous and next values
+          if ((index + 2) === this.workItems.length) {
+            wi.attributes.previousitem = parseInt(this.workItems[index + 1].id);
+          } else {
+            wi.attributes.previousitem = parseInt(this.workItems[index + 2].id);
```

Comment:
This should be the nextitem and the other one should be previousitem. Also you can write the entire logic as the one I mentioned above.